spec/rubykassa/xml_interface_spec.rb

```diff
@@ -31,4 +31,42 @@
   it "correctly transforms method name" do
     expect(subject.send(:transform_method_name, "some_method_name")).to eq("SomeMethodName")
   end
+
+  describe '#base_url' do
+    subject { described_class.new.base_url }
+    context 'sets to test URL when in test mode' do
+      before { Rubykassa.configure { |c| c.mode = :test } }
+      it { is_expected.to eq("http://test.robokassa.ru/Webservice/Service.asmx/") }
+    end
+
+    context 'uses production URL on production' do
+      before { Rubykassa.configure { |c| c.mode = :production } }
+      it { is_expected.to eq("https://merchant.roboxchange.com/WebService/Service.asmx/") }
+    end
+  end
+
+  describe '#op_state' do
+    subject { described_class.new.op_state(params) }
+
+    context 'depends on mode' do
+      let(:params) { { 'StateCode' => 5 } }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.